### DISCUSSION TITLE
Synthea Module Builder support

### DISCUSSION THREAD

**Discussion by andyol74** [synthetichealth/synthea#1470] [Q&A]
We need to create a dataset where, following certain conditions (MedicationOrder and Encounter), Observations are produced. These Observations should be generated periodically (1-2 times per year) within two years from the Encounter or MedicationOrder. We've used the Synthea Module Builder with the WellnessEncounter triggering an Observation. However, the trigger events MedicationOrder and Encounter are too old (> 2 years) compared to the Observations produced. Do you have any suggestions on which components to use in the Builder?

Thanks in advance for your support!

**Comment by jawalonoski** [comment]
If this is all one module, just have a delay state less than two years from the initial wellness encounter. Have the periodic followup encounters be some other encounter class (i.e., **not** wellness), then they'll start as normal. Use loops for the periodic encounters as necessary.

If this is more than one module, set an attribute after the triggering observation, and in the other module, set a Guard that waits for the attribute to have a value, then use a Delay for the amount of time you want, then follow up as above.

Hope that helps.
